pkg/gofr/gofr_test.go

```diff
@@ -504,3 +504,129 @@ func Test_AddCronJob_Success(t *testing.T) {
+	htmlContent := []byte("<html><head><title>Test Static File</title></head><body><p>Testing Static File</p></body></html>")
+
+	err := createPublicDirectory(t, htmlContent)
+	if err != nil {
+		return
+	}
+
+	// Application Generation
+	app := New()
+
+	// Run the Application and compare the fileType and fileSize
+	app.httpRegistered = true
+	app.httpServer.port = 8022
+
+	go app.Run()
+	time.Sleep(1 * time.Second)
+
+	host := "http://localhost:8022"
+
+	tests := []struct {
+		desc                       string
+		method                     string
+		path                       string
+		statusCode                 int
+		expectedBody               string
+		expectedBodyLength         int
+		expectedResponseHeaderType string
+	}{
+		{
+			desc:                       "check file content index.html",
+			method:                     http.MethodGet,
+			path:                       "/" + publicDir + "/index.html",
+			statusCode:                 http.StatusOK,
+			expectedBodyLength:         len(htmlContent),
+			expectedResponseHeaderType: "text/html; charset=utf-8",
+			expectedBody:               string(htmlContent),
+		},
+		{
+			desc:       "check public endpoint",
+			method:     http.MethodGet,
+			path:       "/" + publicDir,
+			statusCode: http.StatusNotFound,
+		},
+	}
+
+	for it, tc := range tests {
+		request, err := http.NewRequestWithContext(context.Background(), tc.method, host+tc.path, http.NoBody)
+		if err != nil {
+			t.Fatalf("TEST[%d], Failed to create request. %s", it, err)
+		}
+
+		request.Header.Set("Content-Type", "application/json")
+
+		client := http.Client{}
+
+		resp, err := client.Do(request)
+		if err != nil {
+			t.Fatalf("TEST[%d], Request failed. %s", it, err)
+		}
+
+		bodyBytes, err := io.ReadAll(resp.Body)
+		if err != nil {
+			t.Fatalf("TEST[%d], Failed to read response body. %s", it, err)
+		}
+
+		body := string(bodyBytes)
+
+		assert.NoError(t, err, "TEST[%d], Failed.\n%s", it, tc.desc)
+		assert.Equal(t, tc.statusCode, resp.StatusCode, "TEST[%d], Failed with Status Body.\n%s", it, tc.desc)
+
+		if tc.expectedBody != "" {
+			assert.Contains(t, body, tc.expectedBody, "TEST [%d], Failed with Expected Body. \n%s", it, tc.desc)
+		}
+
+		if tc.expectedBodyLength != 0 {
+			contentLength := resp.Header.Get("Content-Length")
+			assert.Equal(t, strconv.Itoa(tc.expectedBodyLength), contentLength, "TEST [%d], Failed at Content-Length.\n %s", it, tc.desc)
+		}
+
+		if tc.expectedResponseHeaderType != "" {
+			assert.Equal(t,
+				tc.expectedResponseHeaderType,
+				resp.Header.Get("Content-Type"),
+				"TEST [%d], Failed at Expected Content-Type.\n%s", it, tc.desc)
+		}
+
+		resp.Body.Close()
+	}
+}
+
+func createPublicDirectory(t *testing.T, htmlContent []byte) error {
+	t.Helper()
+
+	directory := "./" + publicDir
+	if _, err := os.Stat(directory); err != nil {
+		if err := os.Mkdir("./"+publicDir, 0755); err != nil {
+			t.Errorf("Couldn't create a "+publicDir+" directory, error: %s", err)
+			return err
```

Comment:
If you call t.Fatalf, you can remove the error you are returning in the function signature, then you no longer have to test err when calling it